### PR TITLE
Add unit tests for db/repos/dapps - Closes #1566

### LINKED ISSUE #1566: Add unit test coverage for db/repos/dapps
Parent: #1484

### Expected behavior
We need 100% coverage for db/repos/dapps

### Actual behavior
Currently there is no coverage for this repo.

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?
No unit test coverage for db/repos/dapps

### How did I fix it?
Added test cases thoroughly for every method in db/repos/dapps. And got 100% coverage for that file.

### How to test it?
```
npm mocha test/unit/db/repos/blocks.js
```
### Review checklist

* The PR solves #1566
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: test/unit/db/repos/dapps.js [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+				const params = {
+					name: 'alpha',
+					link: 'link_to_app',
+					transactionId: '1111',
+				};
+
+				return expect(db.dapps.getExisting(params)).to.be.eventually.eql([]);
+			});
+
+			it('should be resolved with proper response if dapp name matches', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: `with proper data`/`with valid data`

File: test/unit/db/repos/dapps.js [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+				});
+
+				expect(result).to.not.be.null;
+				expect(result).to.be.lengthOf(1);
+				expect(result[0]).to.have.all.keys('name', 'link');
+				expect(result[0].name).to.be.eql(dapp.asset.dapp.name);
+				return expect(result[0].link).to.be.eql(dapp.asset.dapp.link);
+			});
+
+			it('should be resolved with proper response if dapp link matches', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: same

File: test/unit/db/repos/dapps.js [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+				});
+
+				expect(result).to.not.be.null;
+				expect(result).to.be.lengthOf(1);
+				expect(result[0]).to.have.all.keys('name', 'link');
+				expect(result[0].name).to.be.eql(dapp.asset.dapp.name);
+				return expect(result[0].link).to.be.eql(dapp.asset.dapp.link);
+			});
+
+			it('should be resolved with empty response if existing transaction id is passed', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: same

File: test/unit/db/repos/dapps.js [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+					name: dapp.asset.dapp.name,
+					link: dapp.asset.dapp.link,
+					transactionId: dapp.id,
+				});
+
+				return expect(result).to.be.empty;
+			});
+		});
+
+		describe('list()', () => {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: consistency - `list` or `list(params)`, globally

File: test/unit/db/repos/dapps.js [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+			it('should use pass correct params to inline sql', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: `should pass correct...`/`should use correct...`

File: test/unit/db/repos/dapps.js [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+					limit: 10,
+					offset: 20,
+				};
+				yield db.dapps.list(params);
+
+				expect(db.query.firstCall.args[0]).to.be.a('string');
+				return expect(db.query.firstCall.args[1]).to.be.eql(params);
+			});
+
+			it('should rejected with error if required param "limit" is not passed', () => {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: `should be rejected`/`should reject`

File: test/unit/db/repos/dapps.js [hunk 7]
```diff
@@ -0,0 +1,357 @@
...
+				return expect(
+					db.dapps.list({
+						// limit: 10,
+						offset: 0,
+					})
+				).to.be.rejectedWith("Property 'limit' doesn't exist.");
+			});
+
+			it('should rejected with error if required param "offset" is not passed', () => {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: same

File: test/unit/db/repos/dapps.js [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
+			it('should resolve if required params "limit" and "offset" is provided', () => {
+				return expect(
+					db.dapps.list({
+						limit: 10,
+						offset: 0,
+					})
+				).to.be.fulfilled;
+			});
+
+			it('should resolve with proper response', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: `with proper data`/`with proper dapp data`

File: test/unit/db/repos/dapps.js [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+					'category',
+					'icon',
+					'transactionId'
+				);
+				expect(result[0].transactionId).to.eql(dapp.id);
+				expect(result[0].name).to.eql(dapp.asset.dapp.name);
+				return expect(result[0].link).to.eql(dapp.asset.dapp.link);
+			});
+
+			it('should resolved with filtered results if param "where" is provided', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: `should be resolved`/`should resolve`, globally

File: test/unit/db/repos/dapps.js [hunk 10]
```diff
@@ -0,0 +1,357 @@
...
+				const result = yield db.dapps.list({
+					limit: 10,
+					offset: 0,
+					where: ["name = 'Unknown'"],
+				});
+
+				return expect(result).to.be.empty;
+			});
+
+			it('should resolved with ascending results if "sortField" and "sortMethod=ASC" is provided', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: same as above + `are provided`

File: test/unit/db/repos/dapps.js [hunk 11]
```diff
@@ -0,0 +1,357 @@
...
+					offset: 0,
+					sortField: 'name',
+					sortMethod: 'ASC',
+				});
+
+				expect(result).to.be.lengthOf(5);
+				return expect(result).to.be.eql(_.orderBy(result, 'name'));
+			});
+
+			it('should resolved with descending results if "sortField" and "sortMethod=DESC" is provided', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: as above

File: test/unit/db/repos/dapps.js [hunk 12]
```diff
@@ -0,0 +1,357 @@
...
+					offset: 0,
+					sortField: 'name',
+					sortMethod: 'DESC',
+				});
+
+				expect(result).to.be.lengthOf(5);
+				return expect(result).to.be.eql(_.orderBy(result, 'name', 'desc'));
+			});
+
+			it('should paginate the result properly', function*() {
```
Comment (4miners) [inline test/unit/db/repos/dapps.js]: can be fragile, as per other review

Comment (nazarhussain) [reply test/unit/db/repos/dapps.js]: Yes it may be fragile but with small set of data of 5 records to be tested, it is not behaving wrong.

**Pull request merged**